ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/NewWallet.hs

```diff
@@ -212,21 +267,37 @@ hasPasswordToggled nw@NewWallet{..} checked = do
 fillWaletParameters :: NewWallet -> IO (Maybe NewWalletParameters)
 fillWaletParameters NewWallet{..} = do
   nwName <- T.strip . fromString <$> QLineEdit.text walletName
+  nwMnemonic <- T.strip . fromString <$> QLineEdit.text walletMnemonic
+  nwFullRestore <- QAbstractButton.isChecked fullRestore
 
   nwHasPassword <- QAbstractButton.isChecked hasPassword
   nwPasswordCandidate <- T.strip . fromString <$> QLineEdit.text password
   nwRepeatPassword <- T.strip . fromString <$> QLineEdit.text repeatPassword
 
+  nwModeIx <- QComboBox.currentIndex modeSelector
   let
-    nameValid = not $ null nwName
-    passwordValid =
-      (not nwHasPassword) || (not (null nwPasswordCandidate) && nwPasswordCandidate == nwRepeatPassword)
-    nwPassword = if nwHasPassword then Just nwPasswordCandidate else Nothing
+    nwMode =
+      if nwModeIx < fromEnum @NewWalletMode minBound || nwModeIx > fromEnum @NewWalletMode maxBound
+         then Nothing
+         else Just $ toEnum nwModeIx
+
+  return $ do
+    mode <- nwMode
+    guard $ not $ null nwName
+    when nwHasPassword $
+      guard $ not (null nwPasswordCandidate) && nwPasswordCandidate == nwRepeatPassword
+    let nwPassword = if nwHasPassword then Just nwPasswordCandidate else Nothing
+
+    nwSpecifier <- case mode of
+      CreateWallet -> do
+        return $ NewWalletName
+      ImportWallet -> do
+        guard $ not $ null nwMnemonic
+        guard $ length (words nwMnemonic) >= 12
```

Comment:
Not sure if this check is needed here. What do you think?

Reply:
I think it won't harm.